addons/mail/static/src/model/model_listener.js

```diff
@@ -30,9 +30,42 @@ export class Listener {
         this.isPartOfUpdateCycle = isPartOfUpdateCycle;
         this.name = name;
         this.onChange = onChange;
-        // Set of last observed elements, for debugging purposes.
+        /**
+         * Set of localIds that have been accessed on model manager between the
+         * last call to `startListening` and `stopListening` with this listener
+         * as parameter.
+         * Each localId has its own way to know the listeners that are observing
+         * it (to be able to notify them if it changes). This set holds the
+         * inverse of that information, which is useful to be able to remove
+         * this listener (when the need arises) from those localIds without
+         * having to verify the presence of this listener on each possible
+         * localId one by one.
+         */
         this.lastObservedLocalIds = new Set();
-        this.lastObservedFields = new Set();
+        /**
+         * Map between localIds and a set of fields on those localIds that have
+         * been accessed on model manager between the last call to
+         * `startListening` and `stopListening` with this listener as parameter.
+         * Each field of each localId has its own way to know the listeners that
+         * are observing it (to be able to notify them if it changes). This map
+         * holds the inverse of that information, which is useful to be able to
+         * remove this listener (when the need arises) from those fields without
+         * having to verify the presence of this listener on each possible field
+         * one by one.
+         */
+        this.lastObservedFieldsByLocalId = new Map();
+        /**
+         * Set of Model that have been accessed with `all()` on model manager
+         * between the last call to `startListening` and `stopListening` with
+         * this listener as parameter.
+         * Each Model has its own way to know the listeners that are observing
+         * it (to be able to notify them if it changes). This set holds the
+         * inverse of that information, which is useful to be able to remove
+         * this listener (when the need arises) from those Model without having
+         * to verify the presence of this listener on each possible Model one by
+         * one.
+         */
+        this.lastObservedAllByModel = new Set();
```

Comment:
All of this is way too long and way too technical.

The cause is granularity of this class concern being too coarse (model-level registry) when we need something more fine (record-level registry).